CRM/Utils/System/DrupalBase.php

```diff
@@ -269,10 +269,8 @@ public function checkPermissionAddUser() {
   /**
    * @inheritDoc
    */
-  public function logger($message) {
-    if (CRM_Core_Config::singleton()->userFrameworkLogging && function_exists('watchdog')) {
```

Comment:
It might not be common but is possible to call this function from outside debug_log_message(), so I think you'd still want the check for userFrameworkLogging, otherwise it would ALWAYS write to drupal watchdog even when you have that setting turned off. Ditto the drupal 8 function above.

Reply:
Maybe we should add the whole check to the UF classes and remove it from CRM_Core_Error? Reduce duplication.

Reply:
Though looking at the code, I'm not sure I want to complicate it at this point. in `CRM_Core_Error` it's got an extra step of checking the `$statics`, which could be left where it is since I'm not sure if it needs to be replicated in each UF class.